core/org.wso2.carbon.core/src/main/java/org/wso2/carbon/core/util/KeyStoreManager.java

```diff
@@ -329,60 +256,17 @@ public KeyStore getKeyStore(String keyStoreName) throws Exception {
     public KeyStoreMetadata[] getKeyStoresMetadata(boolean isSuperTenant) throws SecurityException {
 
         CarbonUtils.checkSecurity();
-        List<KeyStoreMetadata> metadataList = new ArrayList<>();
-
         try {
-            if (!registry.resourceExists(RegistryResources.SecurityManagement.KEY_STORES)) {
-                return new KeyStoreMetadata[0];
-            }
-            Collection keyStoreCollection = (Collection) registry.get(RegistryResources.SecurityManagement.KEY_STORES);
-            String[] keyStoreFullnames = keyStoreCollection.getChildren();
-
-            for (String keyStoreFullname : keyStoreFullnames) {
-                if (RegistryResources.SecurityManagement.PRIMARY_KEYSTORE_PHANTOM_RESOURCE.equals(keyStoreFullname)) {
-                    continue;
-                }
-                metadataList.add(getKeyStoreMetadata(keyStoreFullname, isSuperTenant));
-            }
+            List<KeyStoreMetadata> metadataList = keyStoreRegistryPersistenceManager.listKeyStores(isSuperTenant);
             if (isSuperTenant) {
                 metadataList.add(getPrimaryKeyStoreMetadata());
             }
             return metadataList.toArray(new KeyStoreMetadata[0]);
-        } catch (RegistryException e) {
-            String msg = "Error when getting keyStore metadata.";
-            throw new SecurityException(msg, e);
+        } catch (SecurityException e) {
```

Comment:
Whta are we catching and throwing the same exception?

Reply:
Fixed in the latest changes